rasa/shared/core/domain.py

```diff
@@ -619,6 +627,41 @@ def _initialize_forms(
             `FormAction` which is implemented in the Rasa SDK.
         """
         if isinstance(forms, dict):
+            # Updating the `not_intent` parameter of each slot with `global_not_intent`
+            if session_config.global_not_intent is not None:
+                global_params = session_config.global_not_intent
+                if not isinstance(global_params, list):
+                    global_params = [global_params]
+                for global_value in global_params:
+                    # looping through all the available forms, their slots and their
+                    # all their slot's types and updating the `not_intent` key
+                    # with the `global_not_intent` value.
+                    for form_count, _ in enumerate(list(forms.keys())):
```

Comment:
rather than using enumerate with the keys why not try iterating over `.items()` this will give you immediate access to both the key and the value.